src/ringct/bulletproofs.cc

```diff
@@ -488,265 +664,402 @@ Bulletproof bulletproof_PROVE(const rct::key &sv, const rct::key &gamma)
 
```

Comment:
When `DEBUG_BP` is active, the above gives an error due to `t0` being undefined. Fix:

     rct::key t0 = inner_product(l0, r0);

Reply:
This line didn't exist before, yes. And I'm pretty sure this is correct; the linear vector polynomials `l(X)` and `r(X)` are defined as:

    l(X) = l0 + l1*X
    r(X) = r0 + r1*X

where

    l0 = vector_subtract(aL, z)
    l1 = sL
    r0 = vector_add(aR, z)
    r1 = hadamard(yMN, sR)

as in the code. The inner product is a quadratic function

    <l(X), r(X)> = t0 + t1*X + t2*X^2

where

    t0 = <l0, r0>
    t1 = <l0, r1> + <l1, r0>
    t2 = <l1, r1>

The newly introduced line which corresponds to the first equation above agrees with the other two which are in the code.